cmd/activator/main.go

```diff
@@ -198,8 +199,11 @@ func main() {
 	params := queue.BreakerParams{QueueDepth: breakerQueueDepth, MaxConcurrency: breakerMaxConcurrency, InitialCapacity: 0}
 
 	// Return the number of endpoints, 0 if no endpoints are found.
-	endpointsGetter := func(revID activator.RevisionID) (int32, error) {
-		count, err := resources.FetchReadyAddressCount(endpointInformer.Lister(), revID.Namespace, revID.Name)
+	endpointsGetter := func(rev *v1alpha1.Revision) (int32, error) {
+		// We have to read the private service endpoints in activator
+		// in order to count the serving pod count.
+		sn := names.PrivateService(rev.Name)
```

Comment:
We're sort of peaking through N layers of abstraction here.  Is there any way we can make `endpointsGetter` take in the SKS instead?

Reply:
Can I do this separately? This will require threading in SKS informer, and lots of other  changes.
This is already huge, so preferably I'll clean this other next iteration.
I'll add a todo.

Reply:
Yep and then I'll switch to getting via SKS.